swap_meet/vendor.py

```diff
@@ -1,2 +1,107 @@
+
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        self.inventory = [] if inventory is None else inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else: 
+            return False
+        
+    def get_by_id(self, item_id):
+
+        for item in self.inventory:
+            if item.id == item_id:
+                return item 
+
+        return None
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+
+        if my_item in self.inventory and their_item in other_vendor.inventory:
+            self.inventory.remove(my_item) 
+            self.inventory.append(their_item)
+            other_vendor.inventory.remove(their_item)
+            other_vendor.inventory.append(my_item)
+            return True
+        
+        return False
+    
+    def swap_first_item(self, other_vendor):
+        if not self.inventory or not other_vendor.inventory:
+            return False
+        
+        elif self.inventory and other_vendor.inventory:
```

Comment:
This condition is not needed, since a false evaluation of the prior `if` condition iplies that neither inventory is empty. Therefore, this could be a plain `else`. But really, like other cases where the true `if` beanch returns, we can drop the else entirely and unindent the logic.